doc/specs/stdlib_stats_distribution_uniform.md

```diff
@@ -0,0 +1,379 @@
+
+```fortran
+program demo_shuffle
+    use stdlib_stats_distribution_PRNG, only : random_seed
+    use stdlib_stats_distribution_uniform, only : shuffle
+    implicit none
+    integer :: seed_put, seed_get, i
+    real :: x(10)
+    integer :: n(10)
+    complex :: z(10)
+
+    do i=1, 10
+        n(i) = i
+        x(i) = real(i)
+        z(i) = cmplx(real(i), real(i))
+    end do
+    seed_put = 32165498
+    call random_seed(seed_put, seed_get)    ! set and get current value of seed
+    print *, shuffle(n)                          ! get randomized n
+
+!10   6   9   2   8   1   3   5   7   4
+
+    print *, shuffle(x)                          ! get randomized x
+
+!5.0   10.0   9.0   4.0   3.0   8.0   2.0   1.0   7.0   6.0
+
+    print *, shuffle(z)                          ! get randomized z
+
+!(8.0, 8.0)    (7.0, 7.0)    (4.0, 4.0)    (1.0, 1.0)    (5.0, 5.0)
+!(9.0, 9.0)    (6.0, 6.0)    (3.0, 3.0)    (2.0, 2.0)    (10.0, 10.0)
+
+end program demo_shuffle
+```
+
+## `rvs_uniform` - uniform distribution random variates
+
+### Status
+
+Experimental
+
+### Description
+
+Without argument the function returns a scalar standard uniformly distributed variate U(0,1) of `real` type with single precision on [0,1].
+
+With single argument `scale` of `integer` type the function returns a scalar uniformly distributed variate of `integer` type on [0,scale]. This is the standard Rectangular distribution.
+
+With single argument `scale` of `real` or `complex` type the function returns a scalar uniformly distributed variate of `real` or `complex` type on [0, scale].
+
+With double arguments `loc` and `scale` the function returns a scalar uniformly distributed random variates of `integer`, `real` or `complex` type on [loc, loc + scale] dependent of input type.
+
+With triple arguments `loc`, `scale` and `array_size` the function returns a rank one array of uniformly distributed variates of `integer`, `real` or `complex` type with an array size of `array_size`.
+
+For `complex` type, the real part and imaginary part are independent of each other.
+
+### Syntax
+
+`result = [[stdlib_stats_distribution_uniform(module):rvs_uniform(interface)]]([[loc,] scale] [[[,array_size]]])`
+
+### Class
+
+Elemental function (without the third argument).
+
+### Arguments
+
+`loc`: optional argument has `intent(in)` and is a scalar of type `integer`, `real` or `complex`.
+
+`scale`: optional argument has `intent(in)` and is a scalar of type `integer`, `real` or `complex`.
+
+`array_size`: optional argument has `intent(in)` and is a scalar of type `integer`.
+
+`loc` and `scale` must have the same type and kind when both are present.
+
+### Return value
+
+The result is a scalar or a rank one array, with size of `array_size`, of type `integer`, `real` or `complex` depending on the input type.
+
+### Example
+
+```fortran
+program demo_uniform_rvs
+    use stdlib_stats_distribution_PRNG, only:random_seed
+    use stdlib_stats_distribution_uniform, only:uni=> rvs_uniform
```

Comment:
Minor point -- for clarity in the documentation, IMO it might be better to use the standard function names in the examples, rather than re-naming.

Reply:
I agree

Reply:
Just to show users one kind of usage. Furthermore, `uni` is a more common name for uniform random variate.